lib/application.js

```diff
@@ -73,7 +75,7 @@ module.exports = {
 
     this.service(location, service, {
       // Any arguments left over are other middleware that we want to pass to the providers
-      middleware: args
+      middleware: parts[0]
```

Comment:
I think this would just add everything before the service happens right? Maybe we need a `middleware: { before: [], after: [] }`?